Util/ONNXExport/src/main/java/org/tribuo/util/onnx/ONNXOperator.java

```diff
@@ -0,0 +1,228 @@
+     * @param input The name of the input.
+     * @param outputs The names of the outputs.
+     * @param attributeValues The attribute names and values.
+     * @return The NodeProto.
+     */
+    default public OnnxMl.NodeProto build(ONNXContext context, String input, String[] outputs, Map<String,Object> attributeValues) {
+        return build(context,new String[]{input},outputs,attributeValues);
+    }
+
+    /**
+     * Builds this node based on the supplied inputs and outputs.
+     * Throws {@link IllegalArgumentException} if the number of inputs or outputs is wrong.
+     * @param context The onnx context used to ensure this node has a unique name.
+     * @param inputs The names of the inputs.
+     * @param outputs The names of the outputs.
+     * @return The NodeProto.
+     */
+    default public OnnxMl.NodeProto build(ONNXContext context, String[] inputs, String[] outputs) {
+        return build(context,inputs,outputs,Collections.emptyMap());
+    }
+
+    /**
+     * Builds this node based on the supplied inputs and outputs.
+     * Throws {@link IllegalArgumentException} if the number of inputs, outputs or attributes is wrong.
+     * May throw {@link UnsupportedOperationException} if the attribute type is not supported.
+     * @param context The onnx context used to ensure this node has a unique name.
+     * @param inputs The names of the inputs.
+     * @param outputs The names of the outputs.
+     * @param attributeValues The attribute names and values.
+     * @return The NodeProto.
+     */
+    default public OnnxMl.NodeProto build(ONNXContext context, String[] inputs, String[] outputs, Map<String,Object> attributeValues) {
+        int numInputs = getNumInputs();
+        int numOptionalInputs = getNumOptionalInputs();
+        int numOutputs = getNumOutputs();
+        String opName = getOpName();
+        String domain = getOpDomain();
+        Map<String, ONNXAttribute> attributes = getAttributes();
+        Set<String> mandatoryAttributeNames = getMandatoryAttributeNames();
+
+        if ((numInputs != VARIADIC_INPUT) && ((inputs.length < numInputs) || (inputs.length > numInputs + numOptionalInputs))) {
+            throw new IllegalArgumentException("Expected " + numInputs + " inputs, with " + numOptionalInputs + " optional inputs, but received " + inputs.length);
+        } else if ((numInputs == VARIADIC_INPUT) && (inputs.length == 0)) {
+            throw new IllegalArgumentException("Expected at least one input for variadic input, received zero");
+        }
+        if (outputs.length != numOutputs) {
+            throw new IllegalArgumentException("Expected " + numOutputs + " outputs, but received " + outputs.length);
+        }
+        if (attributeValues.size() > attributes.size()) {
+            throw new IllegalArgumentException("Found more attributes than expected, received " + attributeValues.size() + ", expected at most " + attributes.size());
```

Comment:
Of all of the attribute correctness checks, this is the one that gives the least informative error message. Since these are short-circuiting, maybe this one should go after the other two.

Reply:
I removed this check as it's redundant given the one immediately after it.